crates/inspection-service/src/inspection_service.rs

```diff
@@ -231,3 +231,109 @@ pub fn start_inspection_service(node_config: NodeConfig) {
             .unwrap();
     });
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use futures::executor::block_on;
+    use hyper::body;
+    use once_cell::sync::Lazy;
+    use prometheus::{register_int_counter, IntCounter};
+    use std::io::read_to_string;
+
+    const INT_COUNTER_NAME: &str = "INT_COUNTER";
+    pub static INT_COUNTER: Lazy<IntCounter> =
+        Lazy::new(|| register_int_counter!(INT_COUNTER_NAME, "An integer counter").unwrap());
+
+    #[test]
+    fn test_inspect_configuration() {
+        let mut config = NodeConfig::default_for_validator();
+        config.inspection_service.expose_configuration = false;
+        let r1f = serve_requests(
+            Request::builder()
+                .uri("http://127.0.0.1:9201/configuration")
+                .method(Method::GET)
+                .body(Body::from(""))
+                .unwrap(),
+            config.clone(),
+        );
+        let mut r1 = block_on(r1f).unwrap();
+        assert_eq!(r1.status(), StatusCode::FORBIDDEN);
+        let r1body = block_on(body::to_bytes(r1.body_mut())).unwrap();
+        assert_eq!(r1body, CONFIGURATION_DISABLED_MESSAGE);
+
+        config.inspection_service.expose_configuration = true;
+        let mut r2 = block_on(serve_requests(
+            Request::builder()
+                .uri("http://127.0.0.1:9201/configuration")
+                .method(Method::GET)
+                .body(Body::from(""))
+                .unwrap(),
+            config,
+        ))
+        .unwrap();
+        assert_eq!(r2.status(), StatusCode::OK);
+        let r2body = block_on(body::to_bytes(r2.body_mut())).unwrap();
+        let r2bs = read_to_string(r2body.as_ref()).unwrap();
+        assert!(r2bs.contains("NodeConfig"));
```

Comment:
nit: I'd declare these strings as constants in the test module (so it's obvious what we're assuming :D)

Reply:
added comments explaining each